discovery_engine_core/core/src/stack.rs

```diff
@@ -87,12 +87,11 @@ impl Stack {
     }
 
     /// Updates the internal documents with the new one and returns an updated [`Stack`].
-    #[allow(dead_code)]
-    pub(crate) fn update<R: Ranker>(
-        mut self,
+    pub(crate) fn update(
+        &mut self,
         new_documents: &[Document],
-        ranker: &R,
-    ) -> Result<Self, Error> {
+        ranker: &impl Ranker,
```

Comment:
With the PR of the client we are returning `Article` from the api and we will use that to construct `Documents`.
I was thinking that we can do that in the `update` function instead after we already filter out some of them so we have run smbert on less documents.

Reply:
makes sense, although i'm not sure how this relates to this pr? i didn't touch the logic of `update()` here (just changed the receiver from `mut self` to `&mut self` as discussed beforehand).